web/src/main/java/org/fao/geonet/services/metadata/MetadataCacheSeeder.java

```diff
@@ -0,0 +1,397 @@
+ *           </list>
+ *       </property>
+ *   </bean>
+ *
+ * @author pmauduit
+ *
+ */
+public class MetadataCacheSeeder extends QuartzJobBean implements ApplicationContextAware {
+
+    /**
+     * The path on the file system where the files can be copied.
+     */
+    private String cachePath;
+    
+    /**
+     * The XSL stylesheet to be used, in the formatter context,
+     * basically the name of the directory containing the view.xsl stylesheet.
+     */
+    private String xslStylesheet;
+
+    /**
+     * The full path to the XSL stylsheet.
+     */
+    private String xslPath;
+    
+    /**
+     * The prefix where the files would be publicly available.
+     * (e.g. http://ids.pigma.org/mds for http://ids.pigma.org/mds/[uuid].html)
+     */
+    private String urlPath = "http://ids.pigma.org/mds";
+    
+    /**
+     * Default change frequency for the sitemap items.
+     */
+    private String changeFreq = "daily";
+    /**
+     * Flag to control if the process is currently running.
+     */
+    private static AtomicBoolean isRunning = new AtomicBoolean(false);
+    
+    /**
+     * The namespace to be used for generating the sitemap.
+     */
+    private final String SITEMAP_NS = "http://www.google.com/schemas/sitemap/0.84";
+    
+    
+    /**
+     * The spring application context, used to get a hook to necessary beans of the running catalogue.
+     */
+    private ApplicationContext applicationContext;
+    
+    /**
+     * Flag indicating if the service is enabled or not.
+     */
+    private boolean enabled = true;
+    
+    /**
+     * Constructor
+     */
+    public MetadataCacheSeeder() {}
+
+    /**
+     * The main entry point of the scheduler.
+     */
+    @Override
+    protected void executeInternal(JobExecutionContext context) throws JobExecutionException {
+
+        if (! enabled) {
+            return;
+        }
+        
+        if (isRunning.get()) {
+            Log.info(Geonet.CACHESEEDER, "Cache seeder already running. Skipping execution.");
+            return;
+        }
+
+        if (applicationContext == null) {
+            Log.error(Geonet.CACHESEEDER,"Application context is null. Be sure to configure SchedulerFactoryBean job factory property with AutowiringSpringBeanJobFactory.");
+            return;
+        }
+
+        ResourceManager resourceManager = applicationContext.getBean(ResourceManager.class);
+        Dbms dbms = null;
+        
+        // It's actually pretty hard to get a hook on jeeves managed objects from a spring object.
+        // Bypassing GN internals and doing it by hand ...
+        // Anyway, the system variable is present on every georchestra's geonetwork ...
+        String userXslDir = System.getProperty(GeonetworkDataDirectory.GEONETWORK_DIR_KEY) + File.separator + "data/user_xsl/";
+        xslPath = userXslDir + xslStylesheet + "/view.xsl";
+
+        try {
+            isRunning.set(true);
```

Comment:
Good precaution but I would expect that the Quartz should ensure that it is only running once (assuming it is correctly configured).

Still a good precaution and doesn't hurt to have it